pkg/app/context.go

```diff
@@ -1059,19 +1059,57 @@ func (ctx *RequestContext) Body() ([]byte, error) {
 type ClientIP func(ctx *RequestContext) string
 
 var defaultClientIP = func(ctx *RequestContext) string {
-	RemoteIPHeaders := []string{"X-Real-IP", "X-Forwarded-For"}
-	for _, headerName := range RemoteIPHeaders {
-		ip := ctx.Request.Header.Get(headerName)
-		if ip != "" {
-			return ip
+	RemoteIPHeaders := []string{"X-Forwarded-For", "X-Real-IP"}
+	TrustedProxies := []string{"0.0.0.0"}
+
+	remoteIP, _, err := net.SplitHostPort(strings.TrimSpace(ctx.RemoteAddr().String()))
+	if err != nil {
+		return ""
+	}
+	trusted := isTrustedProxy(TrustedProxies, remoteIP)
+
+	if trusted {
+		for _, headerName := range RemoteIPHeaders {
+			ip, valid := validateHeader(TrustedProxies, ctx.Request.Header.Get(headerName))
+			if valid {
+				return ip
+			}
 		}
 	}
 
-	if ip, _, err := net.SplitHostPort(strings.TrimSpace(ctx.RemoteAddr().String())); err == nil {
-		return ip
+	return remoteIP
+}
+
+// isTrustedProxy will check whether the IP address is included in the trusted list according to TrustedProxies
+func isTrustedProxy(trustedProxies []string, remoteIP string) bool {
```

Comment:
A map is better than slice here for `trustedProxies`?

Reply:
this is a good idea, thx.